File: backend/src/main/java/com/attendance/controller/dto/TokenResponse.java
package com.attendance.controller.dto;

import com.attendance.domain.UserRole;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class TokenResponse {
    private String accessToken;
    private String refreshToken;
    private String tokenType;
    private long expiresIn;
    private boolean mustChangePassword;
    private UserRole role;
}

File: flutter/lib/core/models/token_response.dart
class TokenResponse {
  const TokenResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.tokenType,
    required this.expiresIn,
    required this.mustChangePassword,
    required this.role,
  });

  final String accessToken;
  final String refreshToken;
  final String tokenType;
  final int expiresIn;
  final bool mustChangePassword;
  final String role;

  factory TokenResponse.fromJson(Map<String, dynamic> json) {
    return TokenResponse(
      accessToken: json['accessToken'] as String,
      refreshToken: json['refreshToken'] as String,
      tokenType: json['tokenType'] as String? ?? 'Bearer',
      expiresIn: json['expiresIn'] as int? ?? 900,
      mustChangePassword: json['mustChangePassword'] as bool? ?? false,
      role: json['role'] as String? ?? 'EMPLOYEE',
    );
  }
}
